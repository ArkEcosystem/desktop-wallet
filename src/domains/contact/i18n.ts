export const translations: { [key: string]: any } = {
	CONTACTS_PAGE: {
		ADD_CONTACT: "Add Contact",
		ADD_CONTACT_MESSAGE: "Add your most frequent contacts for fast, easy payments",
		TITLE: "My Contacts",
		SUBTITLE: "Manage your frequent contacts",
	},
	CONTACT_FORM: {
		ADD_ADDRESS: "Add address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		DELETE_CONTACT: "Delete Contact",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
	MODAL_SEARCH_CONTACT: {
		TITLE: "Search Contact",
		DESCRIPTION: "Find the contact you need quickly and conveniently",
		PLACEHOLDER: "Search...",
	},
	VALIDATION: {
		MAXLENGTH_ERROR: "The Contact Name should have less than {{maxLength}} characters.",
		CONTACT_NAME_EXISTS: "Contact with name {{name}} already exists.",
		CONTACT_NAME_EXISTS_SUFFIX: "already exists",
	},
};
