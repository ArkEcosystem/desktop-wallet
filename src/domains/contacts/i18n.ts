export const translations: { [key: string]: any } = {
	CONTACT_FORM: {
		ADD_ADDRESS: "Add Address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
};
