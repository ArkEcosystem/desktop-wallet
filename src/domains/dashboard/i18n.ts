export const translations: { [key: string]: any } = {
	FILTER_WALLETS: {
		NETWORK: {
			TITLE: "Filter Network",
			DESCRIPTION: "Select the types of networks",
		},
		WALLETS: {
			TITLE: "Your Wallets",
			DESCRIPTION: "Select the display of your wallets",
		},
		CHART: {
			TITLE: "Portfolio Chart",
			DESCRIPTION: "You can hide this unit if you like",
		},
		TRANSACTION_HISTORY: {
			TITLE: "Transaction History",
			DESCRIPTION: "You can hide this unit if you like",
		},
	},

	TRANSACTION_HISTORY: {
		TITLE: "Transactions History",
		EMPTY_TEXT:
			"This will display the history of your transactions. But you don't have more than one transaction at the moment.",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		IMPORT: "Import",
	},

	DASHBOARD_PAGE: {
		CHART: {
			TITLE: "Portfolio Chart",
			PERCENTAGES_LABEL: "Total portfolio",
		},
	},
};
