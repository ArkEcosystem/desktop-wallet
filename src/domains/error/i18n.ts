export const translations: { [key: string]: any } = {
	APPLICATION: {
		TITLE: "Something went wrong",
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		RELOAD: "Reload",
	},
	OFFLINE: {
		TITLE: "Cannot Connect",
		DESCRIPTION: "Check your internet connection if it doesn't work, check it again.",
	},
};
