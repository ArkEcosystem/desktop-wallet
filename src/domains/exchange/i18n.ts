export const translations: { [key: string]: any } = {
	MODAL_ADD_EXCHANGE: {
		TITLE: "Exchange",
		DESCRIPTION:
			"Buy ARK directly from the ARK Desktop Wallet or through one of 20+ exchanges supplying ARK tokens using BTC, ETH or other trading pairs.",
	},
};
