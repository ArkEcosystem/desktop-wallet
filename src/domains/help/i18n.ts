export const translations: { [key: string]: any } = {
	MODAL_CONTACT_US: {
		TITLE: "Contact Us",
		DESCRIPTION: "We'd be happy to answer your questions",
		FIELD_NAME: "Name",
		FIELD_EMAIL: "Email",
		FIELD_SUBJECT: "Subject",
		FIELD_MESSAGE: "Message",
		MESSAGE_REQUIRED: "Message is Required",
		SUBJECT_REQUIRED: "Subject is Required",

		SUBJECT_OPTION: {
			SECURITY: "Security",
			OTHER: "Other",
		},
	},
};
