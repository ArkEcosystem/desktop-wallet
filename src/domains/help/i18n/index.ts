import enUS from "./locales/en-US";

export default {
	"en-US": enUS,
};
