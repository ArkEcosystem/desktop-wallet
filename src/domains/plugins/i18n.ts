export const translations: { [key: string]: any } = {
	MODAL_INSTALL_PLUGIN: {
		DESCRIPTION: "This plugin needs the following permissions:",
		ITEM_1: "Allows access to the Desktop Wallet alerts",
		ITEM_2: "Allows access to play audio from within the Desktop Wallet",
		ITEM_3: "Allows access to the Desktop Wallet events",
	},
};
