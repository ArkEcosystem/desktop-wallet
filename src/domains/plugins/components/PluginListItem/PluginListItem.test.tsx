import { fireEvent, render } from "@testing-library/react";
import { i18n } from "app/i18n";
import React from "react";
import { I18nextProvider } from "react-i18next";

import { PluginListItem } from "./PluginListItem";

describe("PluginListItem", () => {
	it("should render", () => {
		const plugin = {
			id: "ark-explorer",
			name: "ARK Explorer",
			author: "ARK.io",
			category: "utility",
			rating: 4.2,
			version: "1.3.8",
			size: "4.2 MB",
			isInstalled: false,
		};

		const { asFragment, getByTestId } = render(
			<I18nextProvider i18n={i18n}>
				<table>
					<tbody>
						<PluginListItem plugin={plugin} />
					</tbody>
				</table>
			</I18nextProvider>,
		);

		expect(getByTestId("PluginListItem--ark-explorer")).toHaveTextContent("ARK Explorer");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should trigger install", () => {
		const plugin = {
			id: "ark-explorer",
			name: "ARK Explorer",
			author: "ARK.io",
			category: "utility",
			rating: 4.2,
			version: "1.3.8",
			size: "4.2 MB",
			isInstalled: false,
		};

		const onInstall = jest.fn();

		const { asFragment, getByTestId } = render(
			<I18nextProvider i18n={i18n}>
				<table>
					<tbody>
						<PluginListItem plugin={plugin} onInstall={onInstall} />
					</tbody>
				</table>
			</I18nextProvider>,
		);

		fireEvent.click(getByTestId("PluginListItem__install"));

		expect(onInstall).toHaveBeenCalledTimes(1);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should trigger delete", () => {
		const plugin = {
			id: "ark-explorer",
			name: "ARK Explorer",
			author: "ARK.io",
			category: "utility",
			rating: 4.2,
			version: "1.3.8",
			size: "4.2 MB",
			isInstalled: true,
		};

		const onDelete = jest.fn();

		const { asFragment, getByTestId } = render(
			<I18nextProvider i18n={i18n}>
				<table>
					<tbody>
						<PluginListItem plugin={plugin} onDelete={onDelete} />
					</tbody>
				</table>
			</I18nextProvider>,
		);

		fireEvent.click(getByTestId("dropdown__toggle"));
		fireEvent.click(getByTestId("dropdown__option--0"));

		expect(onDelete).toHaveBeenCalledTimes(0);

		fireEvent.click(getByTestId("dropdown__toggle"));
		fireEvent.click(getByTestId("dropdown__option--1"));

		expect(onDelete).toHaveBeenCalledTimes(1);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render official icon", () => {
		const plugin = {
			id: "ark-explorer",
			name: "ARK Explorer",
			author: "ARK.io",
			category: "utility",
			rating: 4.2,
			version: "1.3.8",
			size: "4.2 MB",
			isInstalled: false,
			isOfficial: true,
		};

		const { asFragment, getByTestId } = render(
			<I18nextProvider i18n={i18n}>
				<table>
					<tbody>
						<PluginListItem plugin={plugin} />
					</tbody>
				</table>
			</I18nextProvider>,
		);

		expect(getByTestId("PluginListItem--ark-explorer")).toHaveTextContent("official-ark-plugin.svg");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render grant icon", () => {
		const plugin = {
			id: "ark-explorer",
			name: "ARK Explorer",
			author: "ARK.io",
			category: "utility",
			rating: 4.2,
			version: "1.3.8",
			size: "4.2 MB",
			isInstalled: false,
			isGrant: true,
		};

		const { asFragment, getByTestId } = render(
			<I18nextProvider i18n={i18n}>
				<table>
					<tbody>
						<PluginListItem plugin={plugin} />
					</tbody>
				</table>
			</I18nextProvider>,
		);

		expect(getByTestId("PluginListItem--ark-explorer")).toHaveTextContent("grant.svg");
		expect(asFragment()).toMatchSnapshot();
	});
});
