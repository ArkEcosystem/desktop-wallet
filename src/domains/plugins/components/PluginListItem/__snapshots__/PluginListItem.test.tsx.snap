// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginListItem should render 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="border-b border-dashed border-theme-neutral-200"
        data-testid="PluginListItem--ark-explorer"
      >
        <td
          class="text-center"
        >
          <svg
            class="w-12 h-12"
          >
            changenow-logo.svg
          </svg>
        </td>
        <td>
          <a
            class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
            href="#"
          >
            ARK Explorer
          </a>
        </td>
        <td>
          <div
            class="flex items-center justify-between pr-16"
          >
            <span>
              ARK.io
            </span>
            <div />
          </div>
        </td>
        <td
          class="py-10"
        >
          PLUGINS.CATEGORY.UTILITY
        </td>
        <td
          class="py-10"
        >
          <div
            class="flex"
            data-testid="ReviewRating"
          >
            <div
              class="relative my-auto"
            >
              <div
                class="absolute overflow-hidden"
                style="width: 80%;"
              >
                <svg
                  class="text-theme-warning-300 w-5"
                >
                  star.svg
                </svg>
              </div>
              <svg
                class="text-theme-neutral-500 w-5"
              >
                star.svg
              </svg>
            </div>
            <span
              class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
            >
              4.2
            </span>
          </div>
        </td>
        <td
          class="py-10"
        >
          v 1.3.8
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="w-16 text-right"
        >
          <button
            class="sc-AxjAm ddJBbC"
            color="primary"
            data-testid="PluginListItem__install"
            type="button"
          >
            Install
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`PluginListItem should render grant icon 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="border-b border-dashed border-theme-neutral-200"
        data-testid="PluginListItem--ark-explorer"
      >
        <td
          class="text-center"
        >
          <svg
            class="w-12 h-12"
          >
            changenow-logo.svg
          </svg>
        </td>
        <td>
          <a
            class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
            href="#"
          >
            ARK Explorer
          </a>
        </td>
        <td>
          <div
            class="flex items-center justify-between pr-16"
          >
            <span>
              ARK.io
            </span>
            <div>
              <div
                class="sc-AxirZ hPsuTj"
                height="16"
                width="16"
              >
                <svg>
                  grant.svg
                </svg>
              </div>
            </div>
          </div>
        </td>
        <td
          class="py-10"
        >
          PLUGINS.CATEGORY.UTILITY
        </td>
        <td
          class="py-10"
        >
          <div
            class="flex"
            data-testid="ReviewRating"
          >
            <div
              class="relative my-auto"
            >
              <div
                class="absolute overflow-hidden"
                style="width: 80%;"
              >
                <svg
                  class="text-theme-warning-300 w-5"
                >
                  star.svg
                </svg>
              </div>
              <svg
                class="text-theme-neutral-500 w-5"
              >
                star.svg
              </svg>
            </div>
            <span
              class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
            >
              4.2
            </span>
          </div>
        </td>
        <td
          class="py-10"
        >
          v 1.3.8
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="w-16 text-right"
        >
          <button
            class="sc-AxjAm ddJBbC"
            color="primary"
            data-testid="PluginListItem__install"
            type="button"
          >
            Install
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`PluginListItem should render official icon 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="border-b border-dashed border-theme-neutral-200"
        data-testid="PluginListItem--ark-explorer"
      >
        <td
          class="text-center"
        >
          <svg
            class="w-12 h-12"
          >
            changenow-logo.svg
          </svg>
        </td>
        <td>
          <a
            class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
            href="#"
          >
            ARK Explorer
          </a>
        </td>
        <td>
          <div
            class="flex items-center justify-between pr-16"
          >
            <span>
              ARK.io
            </span>
            <div>
              <div
                class="sc-AxirZ fkjLBb"
                height="15"
                width="15"
              >
                <svg>
                  official-ark-plugin.svg
                </svg>
              </div>
            </div>
          </div>
        </td>
        <td
          class="py-10"
        >
          PLUGINS.CATEGORY.UTILITY
        </td>
        <td
          class="py-10"
        >
          <div
            class="flex"
            data-testid="ReviewRating"
          >
            <div
              class="relative my-auto"
            >
              <div
                class="absolute overflow-hidden"
                style="width: 80%;"
              >
                <svg
                  class="text-theme-warning-300 w-5"
                >
                  star.svg
                </svg>
              </div>
              <svg
                class="text-theme-neutral-500 w-5"
              >
                star.svg
              </svg>
            </div>
            <span
              class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
            >
              4.2
            </span>
          </div>
        </td>
        <td
          class="py-10"
        >
          v 1.3.8
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="w-16 text-right"
        >
          <button
            class="sc-AxjAm ddJBbC"
            color="primary"
            data-testid="PluginListItem__install"
            type="button"
          >
            Install
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`PluginListItem should trigger delete 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="border-b border-dashed border-theme-neutral-200"
        data-testid="PluginListItem--ark-explorer"
      >
        <td
          class="text-center"
        >
          <svg
            class="w-12 h-12"
          >
            changenow-logo.svg
          </svg>
        </td>
        <td>
          <a
            class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
            href="#"
          >
            ARK Explorer
          </a>
        </td>
        <td>
          <div
            class="flex items-center justify-between pr-16"
          >
            <span>
              ARK.io
            </span>
            <div />
          </div>
        </td>
        <td
          class="py-10"
        >
          PLUGINS.CATEGORY.UTILITY
        </td>
        <td
          class="py-10"
        >
          <div
            class="flex"
            data-testid="ReviewRating"
          >
            <div
              class="relative my-auto"
            >
              <div
                class="absolute overflow-hidden"
                style="width: 80%;"
              >
                <svg
                  class="text-theme-warning-300 w-5"
                >
                  star.svg
                </svg>
              </div>
              <svg
                class="text-theme-neutral-500 w-5"
              >
                star.svg
              </svg>
            </div>
            <span
              class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
            >
              4.2
            </span>
          </div>
        </td>
        <td
          class="py-10"
        >
          v 1.3.8
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="w-16 text-right"
        >
          <button
            class="sc-AxjAm kwTlRm text-left"
            color="primary"
            type="button"
          >
            <div
              class="text-left"
            >
              <div
                class="relative"
                data-testid="dropdown__toggle"
              >
                <div
                  class="float-right outline-none focus:outline-none"
                >
                  <div
                    class="sc-AxirZ eEbZjj"
                    height="20"
                    width="20"
                  >
                    <svg>
                      settings.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`PluginListItem should trigger install 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="border-b border-dashed border-theme-neutral-200"
        data-testid="PluginListItem--ark-explorer"
      >
        <td
          class="text-center"
        >
          <svg
            class="w-12 h-12"
          >
            changenow-logo.svg
          </svg>
        </td>
        <td>
          <a
            class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
            href="#"
          >
            ARK Explorer
          </a>
        </td>
        <td>
          <div
            class="flex items-center justify-between pr-16"
          >
            <span>
              ARK.io
            </span>
            <div />
          </div>
        </td>
        <td
          class="py-10"
        >
          PLUGINS.CATEGORY.UTILITY
        </td>
        <td
          class="py-10"
        >
          <div
            class="flex"
            data-testid="ReviewRating"
          >
            <div
              class="relative my-auto"
            >
              <div
                class="absolute overflow-hidden"
                style="width: 80%;"
              >
                <svg
                  class="text-theme-warning-300 w-5"
                >
                  star.svg
                </svg>
              </div>
              <svg
                class="text-theme-neutral-500 w-5"
              >
                star.svg
              </svg>
            </div>
            <span
              class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
            >
              4.2
            </span>
          </div>
        </td>
        <td
          class="py-10"
        >
          v 1.3.8
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="py-10"
        >
          4.2 MB
        </td>
        <td
          class="w-16 text-right"
        >
          <button
            class="sc-AxjAm ddJBbC"
            color="primary"
            data-testid="PluginListItem__install"
            type="button"
          >
            Install
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
